ctru-rs/src/services/am.rs

```diff
@@ -56,6 +68,55 @@ impl<'a> Title<'a> {
             Ok(info.assume_init())
         }
     }
+
+    pub fn get_smdh(&self) -> crate::Result<Smdh> {
+        // i have no idea how to make this look better
+        let archive_path_data: [u32; 4] =
+            [self.low_u32(), self.high_u32(), self.mediatype as u32, 0x0];
+        let smdh_path_data: [u32; 5] = [0x0, 0x0, 0x2, u32::from_le_bytes(*b"icon"), 0x0];
+
+        let archive_path = ctru_sys::FS_Path {
+            type_: ctru_sys::PATH_BINARY,
+            size: size_of::<[u32; 4]>() as u32,
+            data: archive_path_data.as_ptr() as *const libc::c_void,
+        };
+        let smdh_path = ctru_sys::FS_Path {
+            type_: ctru_sys::PATH_BINARY,
+            size: size_of::<[u32; 5]>() as u32,
+            data: smdh_path_data.as_ptr() as *const libc::c_void,
+        };
+
+        let _fs = Fs::init();
```

Comment:
ideally I would suggest we return an error that `Fs` wasn't initialized yet here, if possible, rather than initializing when this function is called.

Reply:
Passing a reference to an existing `Fs` instance might be preferable.

Reply:
I'm thinking of adding sdmh loading code in a type-safe and generic way, into `src/services/fs.rs`, with `Archive<ArchiveType>` struct storing inner `ArchiveType` if the archive requires a binary path, otherwise it would store a generic variant that would represent archives which use only text paths. Also looking at the `smdh_path_data`, it sorta looks like a struct with 2 `repr(u32)` enums, and either `repr(u32)` enum that looks *very much* like a C string without null termination with a null bit put at the end of it, or just a C string